src/main/java/org/fungover/thunder/ClientHandler.java

```diff
@@ -1,33 +1,29 @@
 package org.fungover.thunder;
 
-import java.io.IOException;
-import java.net.ServerSocket;
 import java.net.Socket;
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
 
 public class ClientHandler {
     private final List<Socket> clients;
+
     public ClientHandler() {
-         clients = Collections.synchronizedList(new ArrayList<>());
+        clients = new CopyOnWriteArrayList<>();
     }
-    public void handleConnections(ServerSocket serverSocket){
-        while(!serverSocket.isClosed()){
-            Socket connection = null;
-            try {
-                connection = serverSocket.accept();
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
-            if (connection != null) {
-                System.out.println("New client: " + connection.getInetAddress().getHostName());
-                clients.add(connection);
+
+    public void handleConnection(Socket clientSocket) {
+        try {
+            while (!clientSocket.isClosed()) {
+                System.out.println("New client: " + clientSocket.getInetAddress().getHostName());
+                clients.add(clientSocket);
+                clientSocket.close();
```

Comment:
The while-loop here does not really do anything since the serverSocket.accept() call was moved to the Server-class. Also the clientSocket.close() call here will make the client disconnect instantly after connecting.

Right now, it will only run once and disconnect the client. If you only remove clientSocket.close(); the while-loop will run forever.

Reply:
> It looks good if you remove the while-loop in clienthandler. It might have conflicts with #24 though?

Yes, as you say there will probably be alot more method calls between clients.add and socket.close that will handle the outcome of the closing of the socket. I will remove the while loop as you suggest because in this issue and implementation there is no use of looping. A later PR might implement the loop anyway 👍  Thank you!